rosbag2/src/rosbag2/writer.cpp

```diff
@@ -105,4 +148,21 @@ bool Writer::should_split_bagfile() const
     return storage_->get_bagfile_size() > max_bagfile_size_;
   }
 }
+
+void Writer::finalize_metadata()
+{
+  metadata_.bag_size = 0;
+
+  for (const auto & path : metadata_.relative_file_paths) {
+    metadata_.bag_size += rosbag2_storage::FilesystemHelper::get_file_size(path);
+  }
+
+  metadata_.topics_with_message_count.clear();
+  metadata_.topics_with_message_count.reserve(topics_.size());
+
+  for (const auto & topic : topics_) {
+    metadata_.topics_with_message_count.push_back(topic.second);
+  }
```

Comment:
really only a nitpick, but I believe the call to `std::transform` might also apply in the same manner in case we ever change the datatypes for `topics_` or `metadata_.topics_with_message_count`. 
```
std::transform(topics_.begin(), topics_.end(), std::back_inserter(metadata_.topics_with_message_count),
  [](auto pair) {return pair.second;});
```